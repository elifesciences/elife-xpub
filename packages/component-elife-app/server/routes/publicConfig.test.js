const express = require('express')
const supertest = require('supertest')

describe('Client config handler', () => {
  let routes
  const makeApp = () => {
    const app = express()
    routes(app)
    return supertest(app)
  }

  beforeEach(() => {
    jest.mock('config', () => ({
      publicConfig: {
        isPublic: true,
        key: 'value',
      },
      privateConfig: {
        secret: 'password',
      },
    }))
    routes = require('.')
  })

  it('returns javascript', async () => {
    const request = makeApp()
    const response = await request.get('/config.js').expect(200)
    expect(response.headers['content-type']).toEqual(
      expect.stringContaining('javascript'),
    )
  })

  it('returns a response containing public config', async () => {
    const request = makeApp()
    const response = await request.get('/config.js').expect(200)
    expect(response.text).toMatchSnapshot()
  })
})
