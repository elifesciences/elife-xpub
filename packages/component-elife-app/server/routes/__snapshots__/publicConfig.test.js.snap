// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client config handler returns a response containing public config 1`] = `
"
  window.config = {\\"publicConfig\\":{\\"isPublic\\":true,\\"key\\":\\"value\\"}}
"
`;
